runelite-client/src/main/java/net/runelite/client/plugins/mining/MiningPlugin.java

```diff
@@ -247,6 +247,7 @@ public void onGameObjectDespawned(GameObjectDespawned event)
 			{
 				final WorldPoint point = object.getWorldLocation();
 				respawns.removeIf(rockRespawn -> rockRespawn.getWorldPoint().equals(point));
+				client.clearHintArrow();
```

Comment:
This should check that the hint arrow being cleared is at the same point as the despawned essence via `Client#getHintArrowType()` and `Client#getHintArrowPoint()`. It's possible that the hint arrow was set by the client or by another plugin, so we should not presume that this plugin is the only thing setting it.